pandas/core/arrays/sparse.py

```diff
@@ -887,12 +887,18 @@ def fillna(self, value=None, method=None, limit=None):
 
         return self._simple_new(new_values, self._sparse_index, new_dtype)
 
-    def shift(self, periods=1):
+    def shift(self, periods=1, fill_value=None):
 
         if not len(self) or periods == 0:
             return self.copy()
 
-        subtype = np.result_type(np.nan, self.dtype.subtype)
+        if fill_value is None:
+            res_type = np.nan
+            fill_value = self.dtype.na_value
```

Comment:
I don't understand this. Why do you assign `res_type=np.nan`?

Reply:
because it was used in the previous code like:
```subtype = np.result_type(np.nan, self.dtype.subtype)```

Reply:
sure